mithril-aggregator/src/dependency_injection/builder.rs

```diff
@@ -1598,6 +1621,13 @@ impl DependenciesBuilder {
     }
 }
 
+#[cfg(test)]
+impl DependenciesBuilder {
+    pub(crate) fn new_for_test(configuration: Configuration) -> Self {
```

Comment:
I prefer a name that explains what is does rather than a name that says who should use it or in which context. It's just an opinion, the opposite can also be justified.
Something like : new_with_logs_to_stdout